lambdas/db-migration/src/migrations/20200901060125_create_files_granules.ts

```diff
@@ -0,0 +1,48 @@
+import * as Knex from 'knex';
+
+export const up = async (knex: Knex): Promise<void> => knex.schema
+  .createTable('granules', (table) => {
+    table.bigIncrements('cumulusId').comment('Internal Cumulus ID for a granule').primary();
+    table.text('granuleId').comment('Granule ID');
+    table.text('pdrName').comment('PDR associated with the granule');
+    table.integer('collectionCumulusId').references('cumulusId').inTable('collections').notNullable();
+    table.enum('status', ['running', 'completed', 'failed']).comment('Ingest status of the granule');
+    table.text('execution').comment('Step Function Execution link');
+    table.text('cmrLink').comment('Link to granule in the CMR API');
+    table.text('published').comment('Flag that shows if the granule has been published in CMR');
+    table.float('duration').comment('Ingest duration');
+    table.jsonb('error').comment('JSON error object');
+    table.integer('productVolume');
+    table.float('timeToArchive').comment('Number of seconds granule took to archive');
+    table.float('timeToProcess').comment('Number seconds granule took to complete "processing"');
+    table.date('beginningDateTime').comment('Date granule started');
+    table.date('endingDateTime').comment('Date granule completed');
+    table.date('procvessingStartDateTime').comment('Date granule started processing');
+    table.date('processingEndDateTime').comment('Date granule finished processing');
+    table.date('lastUpdateDateTime').comment('Timestap for last update');
+    table.date('productionDateTime').comment('Timestamp for granule production date/time');
+    table.date('timestamp');
+    table.text('provider').comment('Provider granule is associated with');
+    table.unique(['granuleId', 'collectionCumulusId']);
+    table.timestamps(false, true);
+  }).createTable('files', (table) => {
```

Comment:
I would break up this migration into 2 migrations: one for granule and the other for files. With granules coming first, of course. I think it's a bit more readable/maintainable

Reply:
Hmm, that seems ok.    This change seems structurally linked (we wouldn't ever want to be in the granules but no files state), and given the ordered nature of migrations it seemed convenient to use the same timestamp in the same PR and avoid ordering issues on PRs, but that's an admittedly *minor* point. 

I'd be curious what that pattern looks like for users for some of the (presumably) more complicated migrations we'll have in the future, but we can save that for an arch meeting.     

I'll make that update when I'm in this afternoon.

Reply:
530b434e1